site/components/YoutubeEmbed/index.tsx

```diff
@@ -0,0 +1,102 @@
+import { FC, HTMLAttributes, useState } from "react";
+import { toNumber } from "lodash";
+import * as styles from "./styles";
+import Image, { ImageProps } from "next/image";
+
+/**
+ * For more information about the available youtube embed parameters see:
+ * https://developers.google.com/youtube/player_parameters#release_notes_08_23_2018
+ */
+type YoutubeEmbedProps = {
+  videoId: string;
+  autoplay?: boolean;
+  showSubtitles?: boolean;
+  hideControls?: boolean;
+  hideFullscreen?: boolean;
+  posterImg?: {
+    src: ImageProps["src"];
+    alt: string;
+  };
+} & HTMLAttributes<HTMLDivElement>;
+
+/**
+ * A wrapper around a youtube IFrame to provide a poster overlay
+ * @note auto play does not function on mobile devices
+ * see: https://stackoverflow.com/questions/15090782/youtube-autoplay-not-working-on-mobile-devices-with-embedded-html5-player
+ */
+export const YoutubeEmbed: FC<YoutubeEmbedProps> = ({
+  videoId,
+  posterImg,
+  autoplay,
+  showSubtitles,
+  hideControls,
+  hideFullscreen,
+  ...props
+}) => {
+  // If a poster image is defined then track the play behaviour
+  const [playInitated, setPlayInitiated] = useState(!posterImg);
+
+  // Build the src url
+  const src = `https://www.youtube.com/embed/${videoId}?\
+autoplay=${toNumber(!!(autoplay || playInitated))}&\
+controls=${toNumber(!hideControls)}&\
+fs=${toNumber(!hideFullscreen)}&\
+frameborder="0"&\
+iv_load_policy=3&\
+rel=0&\
+cc_load_policy=${toNumber(!!showSubtitles)}`;
```

Comment:
I don't think you need `toNumber()` on any of these, since it's an interpolated string, and these are all booleans.

Reply:
I thought so too but the youtube embed url expects `0` or `1`. 

https://developers.google.com/youtube/player_parameters#Parameters